docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,461 @@
+You should remove the nodes one-at-a-time and verify that each node has been
+removed from the configuration before moving onto the next one, using the
+`await_removal` API:
+
+[source,js]
+--------------------------------------------------
+# Explicit timeout of one minute
+GET /_nodes/node_name/await_removal?timeout=1m
+# Default timeout of 30 seconds
+GET /_nodes/node_name/await_removal
+--------------------------------------------------
+// CONSOLE
+
+The node (or nodes) for whose removal to wait are specified using
+<<cluster-nodes,node filters>> in place of `node_name` here.
+
+A special case of this is the case where there are only two master-eligible
+nodes and you wish to remove one of them. In this case neither node can be
+safely shut down since both nodes are required to reliably make progress, so
+you must first explicitly _retire_ one of the nodes. A retired node still works
+normally, but Elasticsearch will try and transfer it out of the current
+configuration so its vote is no longer required, and will never move a retired
+node back into the configuration. Once a node has been retired and is removed
+from the configuration, it is safe to shut it down. A node can be retired using
+the retirement API:
+
+[source,js]
+--------------------------------------------------
+# Retire node and wait for its removal up to the default timeout of 30 seconds
+POST /_nodes/node_name/retire
+# Retire node and wait for its removal up to one minute
+POST /_nodes/node_name/retire?timeout=1m
+--------------------------------------------------
+// CONSOLE
+
+The node to retire is specified using <<cluster-nodes,node filters>> in place
+of `node_name` here. If a call to the retirement API fails then the call can
+safely be retried. However if a retirement fails then it's possible the node
+cannot be removed from the configuration due to the
+`cluster.master_nodes_failure_tolerance` setting, so verify that this is set
+correctly first. A successful response guarantees that the node has been
+removed from the configuration and will not be reinstated.
+
+A node (or nodes) can be brought back out of retirement using the `unretire`
+API:
+
+[source,js]
+--------------------------------------------------
+POST /_nodes/node_name/unretire
+--------------------------------------------------
+// CONSOLE
+
+The node (or nodes) to reinstate are specified using <<cluster-nodes,node
+filters>> in place of `node_name` here. After being brought back out of
+retirement they may not immediately be added to the configuration.
+
+[float]
+=== Cluster bootstrapping
+
+A major risk when starting up a brand-new cluster is that you accidentally form
+two separate clusters instead of one. This could lead to data loss: you might
+start using both clusters before noticing that anything had gone wrong, and it
+might then be impossible to merge them together later.
+
+To illustrate how this could happen, imagine starting up a three-node cluster
```

Comment:
I wonder if this description is too long and whether we can add it as a note to the end of the subsection

Reply:
I contemplated this. However my experience of discussions about this topic has been that everyone wants something like Consul's `-bootstrap-expect` option and immediately wants to know why we have a different model that is a bit harder to orchestrate. I don't think I've had a conversation about bootstrapping where this didn't come up as one of the first questions. Everyone trusts their orchestrator too much it seems.

I think this is in the right place in the document, but I've moved it into a `NOTE:` to format it as if it is an aside rather than something that's part of the main body of text.